ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -71,36 +77,87 @@ public StateProcessor(ServiceConfig config, ChainStorageClient chainStorageClien
     this.chainStorageClient = chainStorageClient;
   }
 
-  @Subscribe
-  public void onEth2GenesisEvent(Eth2GenesisEvent event) {
-    STDOUT.log(
-        Level.INFO,
-        "******* Eth2Genesis Event detected ******* : "
-            + ((Eth2Genesis) event).getDeposit_root().toString());
+  public void onEth2Genesis(BeaconState initial_state) {
```

Comment:
It seems like you're reverting to the old version of onEth2Genesis event here. We should keep the new version if we can.

Reply:
Not sure what this means. There is no onEth2Genesis event anymore. There is no Eth2Genesis event in the new spec. We detect it ourselves.

Reply:
I mean you're reverting to the old way Eth2GenesisEvent function was handled. The changes in the spec forced us to change this. So I'm just saying keep the changes we made.

Reply:
I think you should look more closely at the method and where it is used.